infrastructure/lambda.tf

```diff
@@ -0,0 +1,31 @@
+resource "aws_lambda_function" "patent_lambda" {
+  filename      = "patent_informer.zip"
+  function_name = "patent-notifier-dynamodb"
+  role          = "arn:aws:iam::467749311079:role/LabRole"
```

Comment:
it is better to use the account id from the data 
[documentation](https://registry.terraform.io/providers/hashicorp/aws/latest/docs/data-sources/caller_identity)

than you can join it with "arn:aws:iam::" and ":role/LabRole"